.github/workflows/build-mac.yml

```diff
@@ -0,0 +1,163 @@
+        with:
+          path: |
+            src/*.tar.*
+            src/*.zip
+          key: ${{ runner.os }}-${{ hashFiles('**/CMakeLists.txt', 'fetch_resource.sh') }}
+
+      - name: Setup cmake
+        uses: jwlawson/actions-setup-cmake@v1.8
+        with:
+          cmake-version: '3.20.x'
+
+      - name: Install System Dependencies
+      # coreutils for nproc
+      # gnu-getopt for brpc
+        run: |
+          brew install automake
+          brew install coreutils
+          brew install gnu-getopt
+  
+      - name: download
+        if: steps.cache-source.outputs.cache-hit != 'true'
+        run: |
+          ./fetch_resource.sh
+
+      - name: build
+        run: |
+          getopt -V
+          echo 'export PATH="/usr/local/opt/gnu-getopt/bin:$PATH"' >> ~/.bash_profile
+          source ~/.bash_profile
+          getopt -V
+          ./pack_other.sh
+
+      - name: Upload Artifact
+        if: ${{ github.event_name == 'push' }}
+        uses: actions/upload-artifact@v2
+        with:
+          path: |
+            libother-*.tar.gz
+            src/apache-zookeeper-*.tar.gz
+
+  build-llvm:
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Cache Source
+        id: cache-source
+        uses: actions/cache@v2
+        with:
+          path: |
+            src/*.tar.*
+            src/*.zip
+          key: ${{ runner.os }}-${{ hashFiles('**/CMakeLists.txt', 'fetch_resource.sh') }}
+
+      - name: Setup cmake
+        uses: jwlawson/actions-setup-cmake@v1.8
+        with:
+          cmake-version: '3.20.x'
+      
+      - name: Install System Dependencies
+      # coreutils for nproc
+        run: |
+          brew install coreutils
+
+      - name: download
+        if: steps.cache-source.outputs.cache-hit != 'true'
+        run: |
+          ./fetch_resource.sh
+
+      - name: build llvm
+        run: |
+          ./pack_llvm.sh
+
+      - name: Upload Artifact
+        if: ${{ github.event_name == 'push' }}
+        uses: actions/upload-artifact@v2
+        with:
+          path: llvm-*.tar.gz
+
+  build-boost:
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Cache Source
+        id: cache-source
+        uses: actions/cache@v2
+        with:
+          path: |
+            src/*.tar.*
+            src/*.zip
+          key: ${{ runner.os }}-${{ hashFiles('**/CMakeLists.txt', 'fetch_resource.sh') }}
+      
+      - name: Install System Dependencies
+      # coreutils for nproc
+        run: |
+          brew install coreutils
+
+      - name: download
+        if: steps.cache-source.outputs.cache-hit != 'true'
+        run: |
+          ./fetch_resource.sh
+
+      - name: build boost
+        run: |
+          ./pack_boost.sh
+
+      - name: Upload Artifact
+        if: ${{ github.event_name == 'push' }}
+        uses: actions/upload-artifact@v2
+        with:
+          path: boost-*.tar.gz
+
+  release:
```

Comment:
`build-mac`  workflow should merge to the linux one, since it better have on `release` job

when merging, give a different name for `update artifact` on Mac, then download linux & Mac artifacts during 
`release`
u can test on ur fork repo. I did not think so well before

Reply:
Ok, so we will get boost(2), libother(2), llvm(2), thirdparty(2), and only one thirdsrc pkg. Is it fine?

Reply:
yep. boost/llvm/other is fine to remove from release list, they are intermediate artifacts

Reply:
well, same artifact name should fine. just keep the uploaded archive name different from mac & linux 😄

Reply:
OK, we'll release 2 thirdparty pkgs and one thirdsrc pkg.